src/coreclr/runtime.proj

```diff
@@ -6,6 +6,11 @@
     <HasCdacBuildTool Condition="'$(ClrFullNativeBuild)' == 'true' or '$(ClrRuntimeSubset)' == 'true' or '$(ClrDebugSubset)' == 'true' or '$(ClrCrossComponentsSubset)' == 'true'">true</HasCdacBuildTool>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <!-- If changing API level, the value in eng/native/build-commons.sh also needs changed. -->
+    <AndroidApiLevelMin Condition="'$(TargetsAndroid)' == 'true' and '$(UseNativeAotCoreLib)' != 'true'">28</AndroidApiLevelMin>
```

Comment:
@janvorli said that the TLS is the only problem with the lower API level: https://github.com/dotnet/runtime/pull/110471/#discussion_r1919152012

Now that you have ifdefed out the TLS optimization, do we still need this override of API level for CoreCLR?

Reply:
I forgot he fixed it. I validated 21 works and just need to undo some changes.